### ISSUE TITLE
cannot use from_dataframe with a Pandas column of strings. 

### ISSUE THREAD

**Issue by jrivers96** [Paradigm4/SciDB-Py#80]
Using "from_dataframe(A, instance_id=0, **kwargs)" operating off a Pandas column of strings returns type |08 and errors because that type is not included in SDB_NP_TYPE_MAP.

This does not trigger the heirarchy check for string. 

def _sdb_type(np_type):
    if np.issubdtype(np_type, np.character):
        return 'string'

```
if np_type in NP_SDB_TYPE_MAP:
    return NP_SDB_TYPE_MAP[np_type]

raise TypeError("Numpy dtype has no SciDB equivalent: %s" % np_type)
```

SDB_IND_TYPE = 'int64'

**Comment by jrivers96** [comment]
Hi Chris, I am wondering  your intent with from_dataframe? Was it built to support having string columns?

**Comment by ChrisBeaumont** [comment]
String columns are a bit tricky. By default pandas usually packs them in
generic "python object" containers instead of string specific ones. We have
no hope of being able to convert a fully generic object container to a
scidbarray (it could contain anything: floats, date times, functions, etc).

However, you can convert a generic object pandas data frame to a
string-specific one (if it only contains strings). I think scidbpy can
handle that now.

So the two options are to try to auto convert object columns to string
columns inside from_dataframe, or give a more helpful error message that
the user should do this first. The latter is safer, since generic object
containers will not always contain strings

**Comment by jrivers96** [comment]
I don't think you can have a string column unfortunately. When you work with Pandas Dataframes, it always records the dtypes as generic objects.

**Comment by ChrisBeaumont** [comment]
Ah I stand corrected -- you're right. Maybe the best option is to add a `coerce_strings` kwarg to this function that, if true, attempts to convert Series with object dtypes into numpy arrays of string dtype, before converting to SciDBArrays.

**Comment by rvernica** [comment]
Fixed in SciDB-Py `16.9.1`. When uploading NumPy instances where strings are stored as objects an upload schema can be provided.
